### PR TITLE
Frontend for the token-wrapper app

### PR DESCRIPTION
**Pull request by eternauta1337**
This is now working, having tested the following actions:
- Locking and unlocking tokens from an account, displaying correct token vs wrapped token balances.
- Being unable to create a vote or vote from an account with no org tokens.
- Being able to create a vote and vote from an account that has locked tokens and hence has org tokens.

Pls consider that this is my first UI for an app, and I based this work on the guide on hack, so it might not be the way we do things now.

### CODE REVIEW

File: app/index.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<html>
+    <head>
+        <title>Aragon App</title>
```
Comment (sohkai) [inline app/index.html]: Token Wrapper

File: manifest.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "Token Wrapper",
+  "author": "Aragon Association",
```
Comment (sohkai) [inline manifest.json]:
```suggestion
  "author": "Aragon One",
```

File: manifest.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  "description": "Locks/releases ERC20 tokens to mint/burn organization tokens",
+  "details_url": "/meta/details.md",
+  "source_url": "https://<placeholder-repository-url>",
```
Comment (sohkai) [inline manifest.json]:
```suggestion
  "source_url": "https://github.com/aragonone/token-wrapper",
```

File: app/src/abi/ERC20.json [hunk 4]
```diff
@@ -0,0 +1 @@
+../../../build/contracts/ERC20.json
```
Comment (sohkai) [inline app/src/abi/ERC20.json]: Let's just copy the ABI we need, similar to https://github.com/aragon/aragon-apps/blob/master/apps/finance/app/src/abi/token-balanceof.json

Older apps usually have placed separate function ABIs in separate files, but for simplicity we could just group the ones we need into a single file here.

File: contracts/Template.sol [hunk 5]
```diff
@@ -0,0 +1,99 @@
+/*
```
Comment (sohkai) [inline contracts/Template.sol:1]: Don't have to do this now, but we could update this once the template is updated to use our new template structure (https://github.com/aragon/aragon-react-boilerplate/pull/69 should be done soon)

File: app/src/App.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
+import React from 'react'
+import { useAragonApi } from '@aragon/api-react'
+import { Main, Button } from '@aragon/ui'
+import styled from 'styled-components'
+import BN from 'bn.js'
+
+function toBigNumber(value) {
+  /* outputs `value * 10^18` */
+  return (new BN(`${value}`)
```
Comment (sohkai) [inline app/src/App.js]: We probably don't need this string coercion; inputs will just give us strings. Unless we're doing some quick maths on numbers that don't require especially precise calculations for a UI (e.g. calculating percentages), we generally keep numbers as either strings or BNs.

File: package.json [hunk 7]
```diff
@@ -5,11 +5,19 @@
...
+    "start:app": "cd app && npm start && cd ..",
+    "start:ipfs": "./scripts/run.sh ipfs",
+    "start:http": "./scripts/run.sh http",
     "ganache-cli:test": "./node_modules/@aragon/test-helpers/ganache-cli.sh"
   },
   "devDependencies": {
+    "@aragon/cli": "^6.0.0",
     "@aragon/apps-shared-minime": "1.0.0",
     "@aragon/apps-shared-migrations": "1.0.0",
     "@aragon/apps-agent": "1.0.1",
```
Comment (sohkai) [inline package.json]: I think we can remove the `kits-base` dependency, and perhaps some of the unused apps as well.

File: app/src/script.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+  return await app.call('token').toPromise()
+}
+
+async function getERC20() {
+  return await app.call('erc20').toPromise()
+}
+
+async function getTokenBalance(token, account) {
+  const tokenContract = app.external(token, ERC20.abi)
+  return await tokenContract.balanceOf(account).toPromise()
```
Comment (sohkai) [inline app/src/script.js]: The `await`s here (and above) are unnecessary, since the functions are already marked as `async`.

**Review by sohkai — APPROVED** [review]
Some minor details, but LGTM!

**Comment by eternauta1337** [comment]
Thanks for the review @sohkai! All suggestions implemented, merging...

**Pull request merged**